diff --git a/config/webpackDevServer.config.js b/config/webpackDevServer.config.js
index 522a81b9b2b2723822513390a9d7e0fc38f75066..a5c37c7b549e635c46b8892898e4c69e53dd9b89 100644
--- a/config/webpackDevServer.config.js
+++ b/config/webpackDevServer.config.js
@@ -24,2 +24,3 @@ const sockPort = process.env.WDS_SOCKET_PORT;
 module.exports = function (proxy, allowedHost) {
+  const httpsConfig = getHttpsConfig();
   const disableFirewall =
@@ -102,2 +103,7 @@ module.exports = function (proxy, allowedHost) {
-    https: getHttpsConfig(),
+    server: httpsConfig
+      ? {
+          type: 'https',
+          options: httpsConfig === true ? {} : httpsConfig,
+        }
+      : 'http',
     host,
@@ -109,5 +115,4 @@ module.exports = function (proxy, allowedHost) {
     },
-    // `proxy` is run between `before` and `after` `webpack-dev-server` hooks
     proxy,
-    onBeforeSetupMiddleware(devServer) {
+    setupMiddlewares(middlewares, devServer) {
       // Keep `evalSourceMapMiddleware`
@@ -121,4 +126,3 @@ module.exports = function (proxy, allowedHost) {
       }
-    },
-    onAfterSetupMiddleware(devServer) {
+
       // Redirect to `PUBLIC_URL` or `homepage` from `package.json` if url not match
@@ -132,2 +136,4 @@ module.exports = function (proxy, allowedHost) {
       devServer.app.use(noopServiceWorkerMiddleware(paths.publicUrlOrPath));
+
+      return middlewares;
     },
