diff --git a/config/webpackDevServer.config.js b/config/webpackDevServer.config.js
index 522a81b9b2b2723822513390a9d7e0fc38f75066..a5c37c7b549e635c46b8892898e4c69e53dd9b89 100644
--- a/config/webpackDevServer.config.js
+++ b/config/webpackDevServer.config.js
@@ -24,0 +25 @@ module.exports = function (proxy, allowedHost) {
+  const httpsConfig = getHttpsConfig();
@@ -102 +103,6 @@ module.exports = function (proxy, allowedHost) {
-    https: getHttpsConfig(),
+    server: httpsConfig
+      ? {
+          type: 'https',
+          options: httpsConfig === true ? {} : httpsConfig,
+        }
+      : 'http',
@@ -110 +115,0 @@ module.exports = function (proxy, allowedHost) {
-    // `proxy` is run between `before` and `after` `webpack-dev-server` hooks
@@ -112 +117 @@ module.exports = function (proxy, allowedHost) {
-    onBeforeSetupMiddleware(devServer) {
+    setupMiddlewares(middlewares, devServer) {
@@ -122,2 +127 @@ module.exports = function (proxy, allowedHost) {
-    },
-    onAfterSetupMiddleware(devServer) {
+
@@ -132,0 +137,2 @@ module.exports = function (proxy, allowedHost) {
+
+      return middlewares;
